src/Form/DeliveryType.php

```diff
@@ -61,7 +61,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
             // When this is a new delivery,
             // set defaults for pickup/dropoff date
-            if (null === $delivery->getId()) {
+            if (null === $delivery->getId() && true === $options['asap_timing']) {
```

Comment:
That was the reason why the tasks' after/before time from the recurrence rule were not properly displayed